### DISCUSSION TITLE
Hide paths in code

### DISCUSSION THREAD

**Discussion by AlexRMU** [sveltejs/kit#5082] [Q&A] [answered]
In the script "start.js" there is an object with all paths. This partially negates efforts to hide private paths.
I understand that it is possible to obfuscate the code in various ways, but if I want to hide only this section? It would be nice to have this out of the box.
![image](https://user-images.githubusercontent.com/61969794/170449236-b9de2128-7aed-4a7e-bbad-bd4c6a7612a5.png) (adapter-node)
This also applies to variables
![image](https://user-images.githubusercontent.com/61969794/170721748-c80cd914-1d48-4542-9302-02a2adc3dab0.png)

**Comment by itssumitrai** [comment, marked as answer]
Isn't this true for any client side router ? Do you know any client side router which hides the routes in the JS code ? 🤔

**Comment by tnzk** [comment]
How about adding a dynamic route e.g. `/secret/[secret]` and let it serve only when the `secret` is some specific value like `60d660e732057ab88f140c74b0e4a5bd`? This way the object will contain the route itself but the secret value.

You would want to this check to happen in server side (in a page endpoint or load function) if it should be really 'private' anyway.

**Reply by itssumitrai** [reply]
If you want the route to not be present on client side router at all, then I would suggest not putting the route as part of sveltekit. You can put this route outside of svelte-kit as part of your custom server (since you are using adapter-node). Either that or there needs to be a way on Sveltekit to create these `server only` routes which are not included by the client side router, though IMO it seems unnecessary as its straightforward to achieve outside of sveltekit

**Comment by AlexRMU** [comment]
See also https://github.com/sveltejs/kit/pull/13379
